### ISSUE TITLE
"Buffer is not defined" error

### ISSUE THREAD

**Issue by devhero** [marmelab/react-admin#2027]
I set up a react app and followed the tutorial step by step till the paragraph **Displaying A List Of Records**, using the **ra-data-json-server** as dataprovider.

The app compiled successfully but, when the Postlist component should retrieves data from 'http://jsonplaceholder.typicode.com' and renders the list, appeared a red popup with the message "Buffer is not defined".

To be sure I made all the previous steps well I tried the **ra-data-fakerest** library with a fake json, and it worked well.

So I reverted code to use again **ra-data-json-server** and I inspected the thrown error.
I found that the **Buffer** object was undefined.

```
var isValidObject = function isValidObject(value) {
...
  --->  var isBuffer = Buffer && Buffer.isBuffer(value);
...
}
```

After a quick search I found that it's available under node.js but when used in the browser it's missing.

So I added the [buffer](https://github.com/feross/buffer) for browser in the package.json 

```
"dependencies": {
  "buffer": "^5.1.0",
  ...
}
```

and in my index.js I imported it as a global variable before the react-admin initialization

```
import { Buffer } from 'buffer'
window.Buffer = Buffer
```

With this trick I made it works.
Why?

**Environment**
* MacOS: 10.13.5
* React-admin version: 2.1.2
* React version: 16.4.1
* Browser: Firefox 61.0.1, Chrome 67.0
* pnpm version: 2.11.2

**Comment by djhi** [comment]
Closed as duplicate of #1935

**Comment by djhi** [comment]
Some members of the core team are working on MacOs and they do not have this issue. What's your node version ?

**Comment by djhi** [comment]
Besides, the fix in #1935 is not specific to windows

**Comment by devhero** [comment]
I use nvm version: 0.33.8 with node v8.11.3 (LTS: Carbon) and npm v5.6.0

**Comment by fzaninotto** [comment]
The fix in #1935 is wrong, it doesn't solve the problem:

```
Buffer && Buffer.foo
VM119:1 Uncaught ReferenceError: Buffer is not defined
    at <anonymous>:1:1
```

So instead of writing

```js
var isBuffer = Buffer && Buffer.isBuffer(value);
```

We should write

```js
var isBuffer = typeof Buffer !== 'undefined' && Buffer.isBuffer(value);
```

**Comment by fzaninotto** [comment]
that being said, I'm really surprised that we don't experience that bug and you do...
